owncloudApp/src/main/java/com/owncloud/android/presentation/security/passcode/PassCodeActivity.kt

```diff
@@ -194,7 +194,24 @@ class PassCodeActivity : AppCompatActivity(), NumberKeyboardListener, EnableBiom
     private fun setTextListeners() {
         val numberOfPasscodeDigits = (passCodeViewModel.getPassCode()?.length ?: passCodeViewModel.getNumberOfPassCodeDigits())
         for (i in 0 until numberOfPasscodeDigits) {
-            passCodeEditTexts[i]?.setOnClickListener { hideSoftKeyboard() }
+
+            passCodeEditTexts[i]?.apply {
+                hideSoftKeyboard()
+
+                setOnKeyListener { v, keyCode, event ->
+                    if (keyCode == KeyEvent.KEYCODE_TAB && event.action == KeyEvent.ACTION_DOWN) {
+                        val focusDirection = if (passCodeEditTexts.indexOf(v) == 0) {
+                            View.FOCUS_DOWN
+                        } else {
+                            View.FOCUS_LEFT
+                        }
+                        v.focusSearch(focusDirection)?.requestFocus()
+                        true
+                    } else {
+                        false
+                    }
+                }
```

Comment:
Why is all of this inside an `apply` block for each of the `EditText`s? Do you really use `passCodeEditTexts[i]` here? Am I missing something?

Reply:
Yeah, I did it because of the last comment you told me: to unify the `hideSoftKeyboard()` and the `setOnKeyListener` in the same block. But if you agree we the first approach I will remove the `hideSoftKeyboard()` and leave just the `setOnKeyListener` without the `apply`.